### PR TITLE
Config files check

### PR DESCRIPTION
**Pull request by hairyhum**
Changes:

Bugfixes:

Getting filename without extension to check extensions fixed for unix: 
before it was getting everything before the last dot as a name without extension,
now it uses basename and dirname to only strip extension from the file name.
Example:
For `/foo/bar.baz/file` filename without extension was `/foo/bar` and check
for `/foo/bar` == `/foo/bar.baz/file` (that the file has no extension) returned false.

Behaviour changes (script):

If RABBITMQ_ADVANCED_CONFIG_FILE exists, but RABBITMQ_CONFIG_FILE does not,
advanced config will be used as `-config`. This removes confusion if people want
to configure using the erlang terms format.
The script will print a warning to the stdout about that.

If both old and new config files exist and RABBITMQ_CONFIG_FILE points
to both (no extension) - there will be a warinig in the stdout
and the old format config file will be used as `-config`.
New configs will be ignored.

If RABBITMQ_CONFIG_FILE is configured with a wrong extension - there will be an
error and the script will fail.
This might be a breaking change if people are using files like `rabbit.foo.config`
and configure RABBITMQ_CONFIG_FILE to `rabbit.foo`. Should we support that?

If RABBITMQ_ADVANCED_CONFIG_FILE is configured with a wrong extension and is to
be used as `-config` - there will be an error and the script will fail.

Behaviour changes (erlang):

Prelaunch checks that RABBITMQ_ADVANCED_CONFIG_FILE is a valid config file
using `file:consult` and that RABBITMQ_CONFIG_FILE is either a config file
using `file:consult` if it has the `.config` extension or not a config file
if it has the `.conf` extension.
The same thing is checked on server start.

If '.config' file does not exist - it will not be shown in logs.

`-conf_advanced` argument now HAS TO always have extension.
The `.config` extension is always set in scripts.

Refactorings:

Extension check, `-config` argument and generated config arguments (`-conf`, `-conf_advanced`)
and file existense checks now separated.

"Filename without extension" (e.g. RABBITMQ_CONFIG_FILE_NOEX) variables
are set for all config files and used in more places.

More comments in code to explain what's going on.

Defaults:

advanced file now has the .config extension by default, because it can only
have one extension. Configuring it without extension is still supported.

### CODE REVIEW

**Comment by michaelklishin** [comment]
Can we add some bats/shell script tests for this?

File: scripts/rabbitmq-server.bat [hunk 1]
```diff
@@ -48,9 +48,97 @@ if not exist "!ERLANG_HOME!\bin\erl.exe" (
 
 set RABBITMQ_EBIN_ROOT=!RABBITMQ_HOME!\ebin
 
+if not exist "!RABBITMQ_SCHEMA_DIR!" (
```
Comment (lukebakken) [inline scripts/rabbitmq-server.bat]: What do you think about combining the configuration setup code into its own `.bat` file to be shared between `rabbitmq-server.bat` and `rabbitmq-service.bat`?

Comment (hairyhum) [reply scripts/rabbitmq-server.bat]: We do have `rabbitmq-env.bat`. Do you think it can be moved there?

File: scripts/rabbitmq-env [hunk 2]
```diff
@@ -369,6 +545,122 @@ run_escript()
...
+        RABBITMQ_CONFIG_ARG_FILE="$RABBITMQ_CONFIG_FILE"
+    # Config file extension is .conf
+    elif [ "$RABBITMQ_CONFIG_FILE_NOEX.conf" = "$RABBITMQ_CONFIG_FILE" ]; then
+        RABBITMQ_CONFIG_ARG_FILE="$RABBITMQ_ADVANCED_CONFIG_FILE"
+    elif [ "x" != "x$RABBITMQ_CONFIG_FILE" ] && \
+        [ "$RABBITMQ_CONFIG_FILE_NOEX" != "$RABBITMQ_CONFIG_FILE" ]; then
+        # Config file has an extension, but it's neither .conf or .config
+        echo "ERROR: Wrong extension for RABBITMQ_CONFIG_FILE: $RABBITMQ_CONFIG_FILE" 2>&1
+        echo "ERROR: extension should be either .conf or .config" 2>&1
+        return 1
```
Comment (hairyhum) [inline scripts/rabbitmq-env]: This exit should stop the rabbitmq-server sctipt, but it doesn't.
Is there a `set -e` missing anywhere?
I wonder if this check should go to `rabbitmq-server` to keep this file clean from logic.

Comment (lukebakken) [reply scripts/rabbitmq-env]: That is surprising. There is a `set -e` in the `_rmq_env_initialize` function but only if `RABBITMQ_SCRIPTS_DIR` is unset. Let me change the code to check the result of `rabbitmq_config`

Comment (hairyhum) [reply scripts/rabbitmq-env]: Looks fine after 55a695307f8480f5eaf6e01bd0c6edd0112dbdc4

**Comment by hairyhum** [comment]
@lukebakken I keep discovering single closing curly braces like `$RABBITMQ_MNESIA_BASE}/$RABBITMQ_NODENAME`. It feels like there can be more typos like this.
Why are you removing the braces?

**Comment by hairyhum** [comment]
The integration tests are in rabbitmq-ci branch `config_files_check` https://github.com/rabbitmq/rabbitmq-ci/pull/22

File: scripts/rabbitmq-server.bat (hunk 1 above)
Comment (lukebakken) [reply scripts/rabbitmq-server.bat]: Let's wait for future powershell work

**Pull request merged**